os/include/tinyara/fs/automount.h

```diff
@@ -0,0 +1,211 @@
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_NUTTX_AUDIO_AUTOMOUNT_H
+#define __INCLUDE_NUTTX_AUDIO_AUTOMOUNT_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <tinyara/irq.h>
+
+#ifdef CONFIG_FS_AUTOMOUNTER
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+/* Configuration ************************************************************
+ * Automounter configuration
+ *   CONFIG_FS_AUTOMOUNTER - Enables automount support
+ *
+ * Prequisites:
+ *   CONFIG_SCHED_WORKQUEUE -  Work queue support is required
+ *   And others that would only matter if you are working in a very minimal
+ *   configuration.
+ */
+
+/* Helper macros ************************************************************/
+
+#define AUTOMOUNT_ATTACH(s,isr,arg) ((s)->attach(s,isr,arg))
+#define AUTOMOUNT_DETACH(s)         ((s)->attach(s,NULL,NULL))
+#define AUTOMOUNT_ENABLE(s)         ((s)->enable(s,true))
+#define AUTOMOUNT_DISABLE(s)        ((s)->enable(s,false))
+#define AUTOMOUNT_INSERTED(s)       ((s)->inserted(s))
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+/* This is the type of the automount media change handler.  The lower level
+ * code will intercept the interrupt and provide the upper level with the
+ * private data that was provided when the interrupt was attached and will
+ * also provide an indication if the media was inserted or removed.
+ */
+
+struct automount_lower_s; /* Forward reference.  Defined below */
+
+typedef CODE int
+  (*automount_handler_t)(FAR const struct automount_lower_s *lower,
```

Comment:
@sunghan-chang Hello,

I have just copied the header from nuttx, is it OK to modify it's original coding style?